roles/platform/tasks/initialize_setup_aws.yml

```diff
@@ -92,6 +92,12 @@
   ansible.builtin.set_fact:
     plat__aws_subnet_ids: "{{ infra__aws_subnet_ids }}"
 
+- name: Set public subnets for public endpoint access
+  when: plat__public_endpoint_access
+  ansible.builtin.set_fact:
+    plat__aws_public_subnet_ids: "{{ infra__aws_public_subnet_ids }}"
```

Comment:
Where is `infra__aws_public_subnet_ids` defined within the `plat` role?

Reply:
Also, these are "internal" variables for the role, can you drop the `plat` prefix to show that they cannot be modified directly? (That's the intent, anyway, we'll see if we can uphold that goal!)

Reply:
`infra__aws_public_subnet_ids` is a fact set in setup_aws_network.yml#L60. From what i read and verified, facts are shared across plays. 
I stick to the apparent pattern adopted by the code. Variables inside the play used across multiple files had the level prefixed to it like plat__region, plat__namespace, plat__aws_vpc_id, plat__aws_security_group_knox_id... Here the variable is further referenced in https://github.com/guruchai/cloudera.exe/blob/45721e98a5f8856cc0a7245f07e1fed840648a03/roles/platform/tasks/setup_aws_env.yml#L36. 

Let me know if `__aws_public_subnet_ids` is more conformant,  i can change is right away.

Reply:
The pattern I'd like to follow is if the upstream "shared" variable is found, then it should be reassigned, like what you have here. If it is not found, because the upstream role has not been executed, e.g. the use of tags only executes the `platform` role, then this step should go out and discover the values.

Let's get this in and then post a PR to update with the discovery path.